dartagnan/src/main/java/com/dat3m/dartagnan/prototype/expr/BinaryExpression.java

```diff
@@ -0,0 +1,8 @@
+package com.dat3m.dartagnan.prototype.expr;
+
+public interface BinaryExpression extends Expression {
+
+    default Expression getLeft() { return getOperands().get(0); }
+    default Expression getRight() { return getOperands().get(1); }
+
+}
```

Comment:
The `com.dat3m.dartagnan.prototype` package is unused and all but a few of its features (`Free`. `PhiEvent`, `PointerType`) remain uncovered.  Can we remove this before merging?

Reply:
You can remove the whole prototype package. It served its purpose.
And we can always recover the code if needed.